google-cloud-tools-plugin/src/com/google/gct/idea/appengine/cloud/AppEngineDeployAction.java

```diff
@@ -0,0 +1,248 @@
+
+    GeneralCommandLine commandLine = new GeneralCommandLine(
+        appEngineHelper.getGcloudCommandPath().getAbsolutePath());
+    commandLine.addParameters("preview", "app", "deploy", "app.yaml", "--promote", "--quiet");
+    commandLine.addParameter("--version=" + version);
+    commandLine.addParameter("--format=json");
+
+    commandLine.withWorkDirectory(stagingDirectory);
+    consoleLogLn("Working directory set to: " + stagingDirectory.getAbsolutePath());
+
+    try {
+      executeProcess(
+          commandLine,
+          new DeployToAppEngineProcessListener(version));
+    } catch (ExecutionException e) {
+      logger.error(e);
+      callback.errorOccurred(GctBundle.message("appengine.deployment.error.during.execution"));
+    }
+  }
+
+  private File copyFile(File stagingDirectory, String targetFileName, File sourceFilePath)
+      throws IOException {
+    File destinationFilePath = new File(stagingDirectory, targetFileName);
+    FileUtil.copy(sourceFilePath, destinationFilePath);
+    consoleLogLn("Copied %s %s to %s", targetFileName,
+        sourceFilePath.getAbsolutePath(), destinationFilePath.getAbsolutePath());
+    return destinationFilePath;
+  }
+
+  private class DeployToAppEngineProcessListener extends ProcessAdapter {
+    private StringBuilder deployOutput = new StringBuilder();
+    private String version;
+
+    public DeployToAppEngineProcessListener(String version) {
+      this.version = version;
+    }
+
+    @Override
+    public void onTextAvailable(ProcessEvent event, Key outputType) {
+      if(outputType.equals(ProcessOutputTypes.STDOUT)) {
+        deployOutput.append(event.getText());
+      }
+    }
+
+    @Override
+    public void processTerminated(final ProcessEvent event) {
+      if (event.getExitCode() == 0) {
+        callback.succeeded(new DeploymentRuntime() {
+          @Override
+          public boolean isUndeploySupported() {
+            return true;
+          }
+
+          @Override
+          public void undeploy(@NotNull final UndeploymentTaskCallback callback) {
+            SwingUtilities.invokeLater(new Runnable() {
+              @Override
+              public void run() {
+                int doStop = Messages
+                    .showYesNoDialog(
+                        GctBundle.message("appengine.stop.modules.version.confirmation.message"),
+                        GctBundle.message("appengine.stop.modules.version.confirmation.title"),
+                        "OK", "Cancel", General.Warning);
+
+                if (doStop == Messages.YES) {
+                  stop(callback);
+                } else {
+                  callback.errorOccurred("Undeploy Canceled");
+                }
+              }
+            });
+          }
+
+          private void stop(@NotNull UndeploymentTaskCallback callback) {
+            Set<String> modulesToSop;
+            try {
+              modulesToSop = parseDeployOutputToModuleList(deployOutput.toString());
```

Comment:
Can you file a bug to update this code to use the module name configured (once we have module name configuration)?